managers/backup-manager/BackupService.js

```diff
@@ -161,13 +161,14 @@ class BackupService extends BaseDirectorService {
       })
       //TODO-PR - Break it into multiple methods
       .then(backupInfo => {
-        return eventmesh.apiServerClient.updateOperationStateAndResponse({
-            resourceId: opts.instance_guid,
-            operationName: CONST.APISERVER.ANNOTATION_NAMES.BACKUP,
-            operationType: CONST.APISERVER.ANNOTATION_TYPES.BACKUP,
-            operationId: result.backup_guid,
-            stateValue: CONST.APISERVER.RESOURCE_STATE.IN_PROGRESS,
-            response: backupInfo
+        return eventmesh.apiServerClient.updateResource({
```

Comment:
why is update used here and not `patch` method? we should have some guidelines on updating/patching resources across code base, it should not be let to the will of individual developers as it leads to inconsistency and the wrapper methods are then useless.

Reply:
As mentioned above. patch methods (patchResponse, patchOptions) are only used if one wants to patch few more fields in status.respone or spec.options objects as these properties stores stringified json objects, so we first get current value from api server and merge with the new given fields and update resource

Reply:
Amit , I remember you guys maintaining a internal doc -regarding the structure of objects , methods etc.. while developing this feature , can you please add to the wiki ? It will help during code reviews.